### PR TITLE
Ft create articles 159054013

### PR DESCRIPTION
**Pull request by Tittoh**
**What does this PR do?**
Implement CRUD functionality for the articles

**background Information.**
A user can create articles, retrieve, update and delete articles.
Only authenticated users can create, edit and delete articles
Authenticated users can edit and delete their own articles

**How To Manually Test?**
Pull ft-create-articles-159054013
`git ft-create-articles-159054013`

Checkout to ft-create-articles-159054013 branch
`git checkout ft-create-articles-159054013`

Make migrations to create database
`python manage.py makemigrations`
`python manage.py migrate`

Start the server
`python manage.py runserver`

Test the endpoints on postman.

Run test coverage
`coverage run manage.py test`
`coverage report`

**Affected Endpoints:**
`POST` /api/articles/
```
{
  "article": {
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "You have to believe"
  }
}
```
RESPONSE
```
{
    "article": {
        "title": "How to train your dragon",
        "slug": "how-to-train-your-dragon-dd81n6swfw",
        "body": "You have to believe",
        "description": "Ever wonder how?",
        "image_url": null,
        "created_at": "2018-08-09T08:23:17.359497Z",
        "updated_at": "2018-08-09T08:23:17.359631Z",
        "author": {
            "username": "jake",
            "bio": "",
            "image": "https://static.productionready.io/images/smiley-cyrus.jpg"
        }
    }
}
```
`GET` /api/articles/
RESPONSE
```
{
  "articles":[{
    "slug": "how-to-train-your-dragon-jsk3fjs5bg",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg"
    }
  }, 
{
    "slug": "how-to-train-your-dragon-2-d89sfrja7b",
    "title": "How to train your dragon 2",
    "description": "So toothless",
    "body": "It a dragon",
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg"
    }
  }],
  "articlesCount": 2
}
```
`GET` /api/articles/\<slug\>
RESPONSE
```
{
    "article": {
        "title": "How to train your dragon",
        "slug": "how-to-train-your-dragon-dd81n6swfw",
        "body": "You have to believe",
        "description": "Ever wonder how?",
        "image_url": null,
        "created_at": "2018-08-09T08:23:17.359497Z",
        "updated_at": "2018-08-09T08:23:17.359631Z",
        "author": {
            "username": "jake",
            "bio": "",
            "image": "https://static.productionready.io/images/smiley-cyrus.jpg"
        }
    }
}
```
What are the relevant pivotal tracker stories?

[#159054013](https://www.pivotaltracker.com/story/show/159054013) Users should be able to create articles

### CODE REVIEW

**Comment by dessHub** [comment]
I have tested the endpoints and they are working as expected.

File: authors/apps/articles/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (dwarukira) [inline authors/apps/articles/admin.py]: Remove this file as it is not being used.

File: authors/apps/articles/models.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from django.db import models
+from django.utils.text import slugify
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from authors.apps.core.utils import random_string_generator
+from authors.apps.core.models import TimeModel
+
+
+class Article(TimeModel):
+    ''' This class represents the Article model '''
```
Comment (dwarukira) [inline authors/apps/articles/models.py]: Good idea to use double quotes docstring.

**Review by dwarukira — CHANGES_REQUESTED** [review]
Looks good so far. Addressing the changes I requested will make it even better.

File: authors/apps/articles/admin.py (hunk 1 above)
Comment (Tittoh) [reply authors/apps/articles/admin.py]: noted

**Comment by dwarukira** [comment]
Also every time I update anything in the article, not just the title the slug change. Is this the intended behavior.

File: authors/apps/articles/models.py (hunk 2 above)
Comment (Tittoh) [reply authors/apps/articles/models.py]: Let me fix that.

**Comment by asamisellah** [comment]
Include instructions to run migrations in your PR.

**Review by Duncanian — COMMENTED** [review]
I cloned it and it runs well on my local repo. @Tittoh

**Comment by Tittoh** [comment]
I have made the recommended changes. Are there any more reviews?

**Review by dwarukira — APPROVED** [review]
Looks good

**Review by jacksono — APPROVED** [review]

**Pull request merged**